agent/checks/grpc.go

```diff
@@ -49,17 +50,26 @@ func NewGrpcHealthProbe(target string, timeout time.Duration, tlsConfig *tls.Con
 
 // Check if the target of this GrpcHealthProbe is healthy
 // If nil is returned, target is healthy, otherwise target is not healthy
-func (probe *GrpcHealthProbe) Check(target string) error {
+func (probe *GrpcHealthProbe) Check(target string, header map[string][]string) error {
 	serverAndService := strings.SplitN(target, "/", 2)
 	serverWithScheme := fmt.Sprintf("%s:///%s", resolver.GetDefaultScheme(), serverAndService[0])
 
 	ctx, cancel := context.WithTimeout(context.Background(), probe.timeout)
 	defer cancel()
 
+	var headerMD map[string]string = make(map[string]string)
+	for k, v := range header {
+		headerMD[k] = strings.Join(v, ", ")
+	}
+
+	md := metadata.New(headerMD)
+	ctx = metadata.NewOutgoingContext(context.Background(), md)
```

Comment:
Hey @fulviodenza 

Thank you for taking up the `help-wanted` issue, I think a lot of people will appreciate this change! We're fairly close to a release so i'm not sure when we'll be able to get a full review going -- but I did see one of the lint tests failed on this line for ineffectual assignment.

Merging the context timeout line (56) with this one should fix this: 

`ctx, cancel := context.WithTimeout(metadata.NewOutgoingContext(context.Background(), md), probe.timeout)`